### PR TITLE
Button: Properly handle border radius reset

### PR DESCRIPTION
**Pull request by njbrown**
## Description
When the reset button of the border radius slider is clicked, the button in the editor doesn't reflect the change. This bug exists because the `RangeControl` component passed `undefined` to its `onChange` callback when the reset button is hit and this condition was not handled.

This pr fixes it by capturing the initial value of the border radius of the button then resets to it to that value when the reset button is pressed.

## How has this been tested?
Add a Button
Change the value of the border radius
Hit the reset button beside the border radius slider

## Screenshots 
before fix:
roudedbordersbroken

after fix:
roudedbordersfixed

## Types of changes
Bug Fix

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by youknowriad** [comment]
Thanks for the PR @njbrown 

Feels like this issue has been fixed several times already without success :P
Do you know what's the current state here @ItsJonQ and is this a good approach?

**Comment by ItsJonQ** [comment]
@njbrown Thank you for working on this! ❤️

> Do you know what's the current state here @ItsJonQ and is this a good approach?

@youknowriad I took a look. There's something odd about the reset mechanism of this component. I remember helping fix something `RangeControl` related where the expected result (on reset) was `null` rather than `initialPosition`.

There's also a `resetFallbackValue` prop on the component, which appears to be designed to handle this, but it doesn't look like it's used in the code base.

Ideally, the consumer of this component **shouldn't have to manually interpret** change callbacks to manage resets.

Digging through the closed issues.. the previous bug came from `0` handling, rather than reset:
https://github.com/WordPress/gutenberg/pull/22393

---

Looking at the previous implementation 🤔 

https://github.com/WordPress/gutenberg/blob/rnmobile/release-1.12.0/packages/components/src/range-control/index.js#L39

It looks like the reset action caused `onChange()` to be called without any arguments. Technically, the functionality would have been the same before compared to now.

---

As a next step, I'll observe and track issues related to `RangeControl` reset handling to see if I notice continued divergence in how it's implemented.

Does that sound good?

Thanks!

**Comment by youknowriad** [comment]
Yes, that sounds good to me.

**Review by ItsJonQ — APPROVED** [review]
🚀 Looks good to me! Thank you @njbrown !! 

I left a note regarding better reset handling overall. It's not a blocker for this PR.

Approved!

File: packages/block-library/src/button/edit.js [hunk 1]
```diff
@@ -41,9 +41,14 @@ const MAX_BORDER_RADIUS_VALUE = 50;
...
 function BorderPanel( { borderRadius = '', setAttributes } ) {
+	const initialBorderRadius = borderRadius;
 	const setBorderRadius = useCallback(
 		( newBorderRadius ) => {
-			setAttributes( { borderRadius: newBorderRadius } );
+			if ( newBorderRadius === undefined )
+				setAttributes( {
+					borderRadius: initialBorderRadius,
+				} );
+			else setAttributes( { borderRadius: newBorderRadius } );
```
Comment (youknowriad) [inline packages/block-library/src/button/edit.js:51]: It's weird  that  this passes the linting checks, I thought `{}` were enforced.

Comment (ItsJonQ) [reply packages/block-library/src/button/edit.js:51]: Oh, hmm 🤔

Comment (youknowriad) [reply packages/block-library/src/button/edit.js:51]: Not a big deal, but if folks think it could be a good rule we can see if we should add it.

**Pull request merged**